gui/packages/ubuntupro/lib/constants.dart

```diff
@@ -9,3 +9,9 @@ const kDefaultMargin = 32.0;
 
 /// The path of the agent executable relative to the msix root directory.
 const kAgentRelativePath = 'agent/ubuntu-pro-agent-launcher.exe';
+
+/// The full decorated version string
+const kVersion = String.fromEnvironment(
+  'UP4W_FULL_VERSION',
```

Comment:
Unsure to understand, to me, this is taking from environment variable, but at runtime, not build-time, right? Or is this thingy use when compiled?

Reply:
This `.fromEnvironment`  is a confusing name. It means "retrieve the value for the specified name in the compilation configuration environment declaration". It's a compile time constant defined during build via the command line option `--dart-define`.